great_expectations/datasource/data_connector/file_path_data_connector.py

```diff
@@ -124,7 +124,12 @@ def get_batch_definition_list_from_batch_request(
             A list of BatchDefinition objects that match BatchRequest
 
         """
-        batch_request_base: BatchRequestBase = cast(BatchRequestBase, batch_request)
+        # <WILL> TODO 202103 some methods still call this method with BatchRequestBase. Refactor to only allow BatchRequest
```

Comment:
Are we casting a BatchRequest to BatchRequestBase here? The comment sort of makes it sound like we want to do the opposite?

Reply:
Thank you for pointing this out.  This took a bit of digging. 

1) With the current implementation of BatchRequestBase, we do want to keep the cast from `BatchRequest` --> `BatchRequestBase`. This is because want to distinguish `BatchRequest`, which is user input, and `BatchRequestBase` which is intended to be used anytime we are passing `BatchRequest`-like data internally in GE. 

2) The comment, however, is out of date and should be removed. What it should say is that we will be renaming `BatchRequestBase` to something more consistent with the rest of our codebase (the issue is with the `Base` part of the name).  But since that is outside the scope of this PR, I'm just keeping the cast `BatchRequest` --> `BatchRequestBase`. I'm really sorry for missing this one... and appreciate you pointing it out.